sys/arm64/arm64/trap.c

```diff
@@ -239,6 +252,35 @@ external_abort(struct thread *td, struct trapframe *frame, uint64_t esr,
 	panic("Unhandled EL%d external data abort", lower ? 0: 1);
 }
 
+#if __has_feature(capabilities)
+static void
+cap_abort(struct thread *td, struct trapframe *frame, uint64_t esr,
+    uint64_t far, int lower)
+{
+	struct pcb *pcb;
+
+	pcb = td->td_pcb;
+	if (!lower) {
+		if (td->td_intr_nesting_level == 0 &&
+		    pcb->pcb_onfault != 0) {
+			frame->tf_x[0] = EPROT;
+			trapframe_set_elr(frame,
+			    (uintcap_t)cheri_setaddress(cheri_getpcc(),
+			    pcb->pcb_onfault));
+			return;
+		}
+		print_registers(frame);
+		printf(" far: %16lx\n", far);
+		printf(" esr:         %.8lx\n", esr);
+		panic("Capability abort from kernel space!");
+	}
+
+	call_trapsignal(td, SIGPROT, 0, (void * __capability)frame->tf_elr,
```

Comment:
I guess a later change pulls in the `si_code` for `SIGPROT`?  I would kind of expect that helper function for `si_code` to be included here and for `si_code` to be correct from the start.

Reply:
Hm, yes, that commit belongs here not in #842. There's be a slight annoying rebase of #842 needed once this PR is merged with that in order to resolve the conflicts in files.arm64 and cheri.h but that's fine.

Reply:
Alternatively, you could just move this commit to the end of #842 and patch it to use the function to map the `si_code`.

Reply:
Yeah, perhaps having copyinout in a separate PR just doesn't make sense given it's all related.